File: java/twelve-days/src/main/java/TwelveDays.java
import java.util.stream.IntStream;
import java.util.stream.Collectors;

class TwelveDays {
    private final String[][] verseParts = { null,
        { "first", "a Partridge in a Pear Tree.\n" },
        { "second", "two Turtle Doves, and" },
        { "third", "three French Hens," },
        { "fourth", "four Calling Birds," },
        { "fifth", "five Gold Rings," },
        { "sixth", "six Geese-a-Laying," },
        { "seventh", "seven Swans-a-Swimming," },
        { "eighth", "eight Maids-a-Milking," },
        { "ninth", "nine Ladies Dancing," },
        { "tenth", "ten Lords-a-Leaping," },
        { "eleventh", "eleven Pipers Piping," },
        { "twelfth", "twelve Drummers Drumming," },
    };

    String verse(int verseNumber) {
        return new StringBuilder(
            "On the " + this.verseParts[verseNumber][0] + " day of Christmas my true love gave to me:"
            + IntStream.range(0, verseNumber)
                .mapToObj(i -> " " + verseParts[verseNumber - i][1])
                .collect(Collectors.joining())
        ).toString();
    }

    String verses(int startVerse, int endVerse) {
        return IntStream.rangeClosed(startVerse, endVerse)
            .mapToObj(this::verse)
            .collect(Collectors.joining("\n"));
    }
    
    String sing() {
        return verses(1, 12);
    }
}

File: python/twelve-days/twelve_days.py
SONG_PARTS = [ None,
  ['first'],
  ['second', 'two Turtle Doves, and '],
  ['third', 'three French Hens, '],
  ['fourth', 'four Calling Birds, '],
  ['fifth', 'five Gold Rings, '],
  ['sixth', 'six Geese-a-Laying, '],
  ['seventh', 'seven Swans-a-Swimming, '],
  ['eighth', 'eight Maids-a-Milking, '],
  ['ninth', 'nine Ladies Dancing, '],
  ['tenth', 'ten Lords-a-Leaping, '],
  ['eleventh', 'eleven Pipers Piping, '],
  ['twelfth', 'twelve Drummers Drumming, '],
]

def recite(start_verse, end_verse):
  song = []
  for verse in range(start_verse, end_verse + 1):
    res = f'On the {SONG_PARTS[verse][0]} day of Christmas my true love gave to me: '
    for num in range(verse, 1, -1):
      res += SONG_PARTS[num][1]
    song.append(res + 'a Partridge in a Pear Tree.')
  
  return song